Add Contact Search By Custom Field

peterbybel opened the pull request:
This adds the ability to search for a contact using a custom field and value. It is most useful to provide a single result when searching for a field with a unique identifier. The response is a JSON object that is a bit ambiguous to navigate. To document, the response will look something like: 

```
{
  '0': DATA_FOR_FIRST_RESULT,
  '1': DATA_FOR_SECOND_RESULT,
  '2': DATA_FOR_THIRD_RESULT,
  'result_code': 1,
  'result_message: 'Success: Something is returned',
  'result_output': 'json',
}
```

This cardinal index structure of the response is the reason it might be smelly to navigate. That being said, if you are searching a unique identifier, and expect one and only one result, you can then navigate to the data using something like `response['0']['id']` which would give you the id of the contact returned in the search.

Then, you will likely need to use the `contact_view` or `contact_update` methods to lookup/update the contact.

peterbybel committed:
Add custom_field_search method

I also added some comments above the method to describe some of the gotchas to be aware of in using this method.

clintmiller commented on lib/active_campaign/v2/clients/contact.rb:
```diff
@@ -60,6 +60,27 @@ def contact_list(ids: 'all', **args)
+        # expect there to be one and only one result with a unique identifier, it should work.
+        #
+        # NOTE: Contacts that are not subscribed to any list will not be returned in the result set.
+        def custom_field_search(field:, value:, **args)
```
Would it be helpful to do `field.upcase` or is it reasonable to assume that consumers of this API know that custom fields are ALL_UPCASE?

peterbybel replied:
Great call here. Personally, I'd prefer to pass a symbol, so I included a `to_s` as well to handle it. Now users can submit:

```
:test_field
'test_field'
'TEST_FIELD'
```

And they will be formatted appropriately.

clintmiller reviewed (approved):
Looks good. Can't wait until you need a function that allows filtering by multiple custom fields. 😉

peterbybel committed:
Format field param into upcase string

This will allow users to submit any of the following as valid field params in the search:

1. 'test_field'
2. 'TEST_FIELD'
3. :test_field

peterbybel commented:
> Can't wait until you need a function that allows filtering by multiple custom fields.

I've got my "NO" response on deck!
